Lib/test/test_csv.py

```diff
@@ -1430,5 +1431,11 @@ def test_subclassable(self):
         # issue 44089
         class Foo(csv.Error): ...
 
+    def test_reader_disallow_instantiation(self):
+        check_disallow_instantiation(self, _csv.Reader)
+
+    def test_writer_disallow_instantiation(self):
+        check_disallow_instantiation(self, _csv.Writer)
```

Comment:
This is a CPython only implementation detail, so we should decorate the tests with `@test.support.cpython_only`. I'd also suggest to use the `test.support.import_helper` to import the C extension module instead of doing that toplevel.

```suggestion
    @support.cpython_only
    def test_disallow_instantiation(self):
        _csv = support.import_helper.import_module("_csv")
        for tp in _csv.Reader, _csv.Writer:
            with self.subTest(tp=tp):
                check_disallow_instantiation(self, tp)

```